src/main/java/org/generationcp/middleware/manager/GermplasmDataManagerImpl.java

```diff
@@ -1599,4 +1599,16 @@ public long countAttributesByGid(final String gid, final List<String> attributeD
 		return this.getAttributeDao().getByIDs(ids);
 	}
 
+	@Override
+	public List<Germplasm> getExistingCrosses(final String femaleParent, final int methodId, final List<Integer> maleParentIds,
```

Comment:
Apply for both methods: `femaleParent` and `gid` parameters should be Integer type

Reply:
Make `gid` as `Optional<Integer>`